packages/connect/src/backend/BackendManager.ts

```diff
@@ -67,23 +73,54 @@ export class BackendManager {
     }
 
     setCustom(shortcut: CoinShortcut, blockchainLink: BlockchainLink) {
-        this.removePreferred(shortcut);
+        this.setPreferred(shortcut, undefined);
         this.custom[shortcut] = blockchainLink;
     }
 
     removeCustom(shortcut: CoinShortcut) {
-        this.removePreferred(shortcut);
+        this.setPreferred(shortcut, undefined);
         delete this.custom[shortcut];
     }
 
+    private setInstance(shortcut: CoinShortcut, instance: Blockchain | undefined) {
+        if (!instance) delete this.instances[shortcut];
+        else this.instances[shortcut] = instance;
+    }
+
     // keep backend as a preferred once connection is successfully made
     // switching between urls could lead to side effects (mempool differences, non existing/missing pending transactions)
-    private setPreferred(shortcut: CoinShortcut, url: string) {
-        this.preferred[shortcut] = url;
+    private setPreferred(shortcut: CoinShortcut, url: string | undefined) {
+        if (!url) delete this.preferred[shortcut];
+        else this.preferred[shortcut] = url;
     }
 
-    private removePreferred(shortcut: CoinShortcut) {
-        delete this.preferred[shortcut];
+    private planReconnect(
```

Comment:
ok, so the logic here should be the same is it was previsouly in blockchainThunk

Reply:
Exactly.